code/modules/antagonists/wizard/grand_ritual/grand_ritual.dm

```diff
@@ -0,0 +1,286 @@
+/**
+ * The Grand Ritual is the Wizard's alternate victory condition
+ * and also a tool to make funny distractions and progress the round state.
+ *
+ * The wizard is assigned a random area to perform the ritual in.
+ * This entails travelling to that area, drawing a 3x3 rune, and casting on it for a while.
+ * Completing these causes a random event to immediately occur and may cause additional side effects in the area.
+ * The more rituals are completed, the more dramatic the events which can be spawned.
+ *
+ * After passing certain thresholds Grand Ritual completions will begin spawning active and expended Reality Tears.
+ * Above a certian threshold, beginning the ritual will alert the crew to your location.
+ *
+ * The 7th ritual completion is special and allows you to pick a "finale" effect which should be very dramatic.
+ * Further completion after that returns to the usual behaviour.
+ */
+/datum/action/cooldown/grand_ritual
+	name = "Grand Ritual"
+	desc = "Provides direction to a nexus of power, then draws a rune in that location for completing the Grand Ritual. \
+		The ritual process will take longer each time it is completed."
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_HANDS_BLOCKED
+	background_icon_state = "bg_spell"
+	button_icon = 'icons/mob/actions/actions_cult.dmi'
+	button_icon_state = "draw"
+	cooldown_rounding = 0
+	/// Path to area we want to draw in next
+	var/area/target_area
+	/// Number of times the grand ritual has been completed somewhere by this user
+	var/times_completed = 0
+	/// If you have drawn your finale rune
+	var/drew_finale = FALSE
+	/// True while you are drawing a rune, prevents action spamming
+	var/drawing_rune = FALSE
+	/// Weakref to a rune drawn in the current area, if there is one
+	var/datum/weakref/rune
+
+	/// A blacklist of turfs we cannot scribe on.
+	var/static/list/blacklisted_rune_turfs = typecacheof(list(/turf/closed/indestructible, /turf/open/indestructible, /turf/open/space, /turf/open/openspace, /turf/open/lava, /turf/open/chasm))
+	/**
+	 * Areas where you can place a rune
+	 * To be honest if maintenance subtypes didn't exist I could probably have got away with just a blacklist, c'est la vie
+	 */
+	var/static/list/area_whitelist = typecacheof(list(
+		/area/station/commons,
```

Comment:
alphabetize this list and the one below it. also the one i asked you to multi-line.